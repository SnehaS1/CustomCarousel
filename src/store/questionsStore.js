import { create } from 'zustand';

const questionsList = [{
  id: 0,
  questions: 'How was your overall week ?',
  rating: null,
  backgroundColor: '#BFB2F3'
}, {
  id: 1,
  questions: 'How would you rate your satisfaction with the customer service representative you spoke to?',
  rating: null,
  backgroundColor: '#96CAF7'
}, {
  id: 2,
  questions: 'How would you rate the effort you had to make to get in touch with us?',
  rating: null,
  backgroundColor: '#9CDCAA'
}, {
  id: 3,
  questions: 'How would you rate the effort you had to make to speak directly to someone on the customer service team?',
  rating: null,
  backgroundColor: '#E5E1AB'
}, {
  id: 4,
  questions: 'How would you rate the quality of your customer service experience ?',
  rating: null,
  backgroundColor: '#F8A3A8'
}, {
  id: 5,
  questionsListFinal: [],
  ratings: [],
  backgroundColor: '#ff80bf'
}];

const handleFeedback = (val) => {
  switch (val) {
    case 0:
      return 'Nuetral';
    case 1:
      return 'Like';
    case -1:
      return 'Dislike';
  }
};

export const useQuestionsStore = create((set) => ({
  // initial state
  questionsList: questionsList,
  activeIndex: 0,
  // methods for anipulating state
  setActiveindex: (id) => {
    set(() => ({ activeIndex: id }));
  },
  updateQuestions: (id, val) => {

    set((state) => ({
      questionsList: state.questionsList.map((question) =>
        question.id === id
          ? ({ ...question, rating: handleFeedback(val) })
          : question
      ),
    }));



  },
}));