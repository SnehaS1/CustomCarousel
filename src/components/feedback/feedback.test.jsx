import { act, fireEvent, render, renderHook, screen, within } from "@testing-library/react";
import FeebackConatiner from "./feedback";
import { useQuestionsStore } from "../../store/questionsStore";
import userEvent from '@testing-library/user-event';

describe("Register component", () => {
  beforeEach(() => {
    const { result } = renderHook(() => useQuestionsStore());

    // Reset the state
    act(() => {
      result.current.activeIndex = 0;
    });
  });
  it(`MyComponent renders with default props`, () => {
    const { asFragment } = render(<FeebackConatiner questionId={1} />)

    expect(asFragment(<FeebackConatiner questionId={1} />)).toMatchSnapshot()
  });
  it("should render feedback component correctly", () => {
    render(<FeebackConatiner questionId={1} />);
    const element = screen.getByRole('feedback-section')
    expect(element).toBeInTheDocument();
  });
  it("should render 3 emojis correctly", () => {
    render(<FeebackConatiner questionId={1} />);
    expect(screen.getAllByRole('emoji').length).toBe(3);
  });
  it("should update on click", () => {
    render(<FeebackConatiner questionId={1} />);
    const { result } = renderHook(() => useQuestionsStore());
    expect(result.current.activeIndex).toBe(0);
  });
  it("should update on click", () => {
    render(<FeebackConatiner questionId={1} />);
    const { result } = renderHook(() => useQuestionsStore());
    expect(result.current.questionsList.length).toBe(6);
  });
  it("should update on click", () => {
    render(<FeebackConatiner questionId={1} />);
    const { result } = renderHook(() => useQuestionsStore());
    expect(result.current.questionsList.length).toBe(6);
  });
  it("should update on click", async () => {
    render(<FeebackConatiner questionId={1} />);
    const { result } = renderHook(() => useQuestionsStore());
    expect(result.current.activeIndex).toBe(0);
    // await userEvent.click(screen.getByRole('emoji'), 0);
    await userEvent.click(screen.getAllByRole('emoji')[0])
    expect(result.current.activeIndex).toBe(1);
    // expect(result.current.questionsList.length).toBe(6);

    result.current.setActiveindex
  });
  it('should handle onclick ', async () => {
    render(<FeebackConatiner questionId={1} />);
    const mockedSetTodo = jest.fn();

  })
  it('should update handleEmojiClick', async () => {
    function renderOfficeCharacters() {
      render(<FeebackConatiner questionId={1} />);

      return {
        getEmojis() {
          return screen.getAllByTestId("emoji-list-li").map((item) => ({
            value: within(item).getByTestId("emoji-tooltip").textContent,
            handleEmojiClick: within(item).getByTestId("emoji-wrapper")
          }));
        }
      };
    }
    const { getEmojis } = renderOfficeCharacters();
    const likeVal = "Like";
    expect(
      getEmojis().find((character) => character.value === likeVal)
    ).toBeTruthy();
  })
});
