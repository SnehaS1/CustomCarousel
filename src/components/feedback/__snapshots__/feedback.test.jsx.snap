// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component MyComponent renders with default props 1`] = `
<DocumentFragment>
  <section>
    <ul
      class="emoji-list"
      role="feedback-section"
    >
      <li
        data-testid="emoji-list-li"
        role="emoji"
      >
        <p
          class="emoji-wrapper"
          data-testid="emoji-wrapper"
        >
          <span
            class="emoji"
          >
            👍
          </span>
          <span
            class="tooltip"
            data-testid="emoji-tooltip"
          >
            Like
          </span>
        </p>
      </li>
      <li
        data-testid="emoji-list-li"
        role="emoji"
      >
        <p
          class="emoji-wrapper"
          data-testid="emoji-wrapper"
        >
          <span
            class="emoji"
          >
            🤔
          </span>
          <span
            class="tooltip"
            data-testid="emoji-tooltip"
          >
            Nuetral
          </span>
        </p>
      </li>
      <li
        data-testid="emoji-list-li"
        role="emoji"
      >
        <p
          class="emoji-wrapper"
          data-testid="emoji-wrapper"
        >
          <span
            class="emoji"
          >
            👎
          </span>
          <span
            class="tooltip"
            data-testid="emoji-tooltip"
          >
            Dislike
          </span>
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
